clang/test/CodeGenSYCL/sycl-intelfpga-field-no-opaque.cpp

```diff
@@ -0,0 +1,73 @@
+// RUN: %clangxx %s -fsycl-device-only -fintelfpga -S -o %t.ll
+// RUN: FileCheck %s --input-file %t.ll
+
+// This test checks that proper IR is generated for kernel field initialization, including
+// 3 cases:
+//   1. initialize pointer field with a global pointer
+//   2. initialize a float field
+//   3. initialize a BitInt field
+// It also checks that the kernel lambda is not inlined for FPGA, i.e. the kernel entry
+// contains `call @NameOfCallOperator(...)`
+
+// Note this is a tempoary test for the FPGA-specific use model that will be
+// replaced by kernel argument compile-time properties.
+
+#include "Inputs/sycl.hpp"
+
+
+struct fooA {
+    int *p;
+
+// CHECK: define dso_local spir_kernel void @_ZTS4fooA(i32 addrspace(1)* {{.*}}%[[ARG:.*]])
+// CHECK: %[[ARG_ADDR:.*]] = alloca i32 addrspace(1)*, align 8
+// CHECK: store i32 addrspace(1)* %[[ARG]], i32 addrspace(1)** %[[ARG_ADDR]], align 8
+// CHECK-DAG: %[[LOAD_ARG:.*]] = load i32 addrspace(1)*, i32 addrspace(1)** %[[ARG_ADDR]], align 8
+// CHECK-DAG: %[[ADDR_CAST:.*]] = addrspacecast i32 addrspace(1)* %[[LOAD_ARG]] to i32 addrspace(4)*
+// CHECK-DAG: %[[GEP:.*]] = getelementptr inbounds
+// CHECK: store i32 addrspace(4)* %[[ADDR_CAST]], i32 addrspace(4)** %[[GEP]], align 8
+    fooA(int *_p) : p(_p) {}
+
+
+// CHECK: call spir_func void @_ZNK4fooAclEv(
+// CHECK: call void @__itt_offload_wi_finish_wrapper()
```

Comment:
You want to check the instrumented annotations call here?

Reply:
This is not a concern when changing the lit test command to "clang_cc1', because the instrumented annotation call is not added at this stage.